### PR TITLE
Add `Unpacker::remaining_bytes` method

### PR DESCRIPTION
**Pull request by pvdrz**

### CODE REVIEW

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
What about the io implementation ?

**Comment by pvdrz** [comment]
> What about the io implementation ?

It returns `None` by default.

File: bee-message/src/payload/mod.rs [hunk 1]
```diff
@@ -260,6 +260,18 @@ impl Packable for OptionalPayload {
         let len = u32::unpack::<_, VERIFY>(unpacker).infallible()? as usize;
 
         if len > 0 {
+            if let Some(remaining_bytes) = unpacker.remaining_bytes() {
+                if len > remaining_bytes {
+                    return Err(UnpackError::Packable(
+                        ValidationError::PayloadLengthMismatch {
```
Comment (thibault-martinez) [inline bee-message/src/payload/mod.rs]: Do you think we can benefit from a more precise error ? It's technically not a mismatch, more of a lack of bytes

File: bee-message/tests/message.rs [hunk 2]
```diff
@@ -240,6 +243,25 @@ fn unpack_invalid_version() {
...
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 3, 238, 142, 220, 195, 72, 99, 77, 135,
+        73, 71, 196, 160, 101, 213, 130, 203, 214, 96, 245, 30, 3, 44, 37, 103, 128, 55, 240, 155, 139, 220, 142, 178,
+        216, 230, 192, 191, 209, 104, 112, 20, 2, 0, 0, 0, 109, 0, 0, 0, 0, 0, 0, 0, 0, 145, 167, 69, 239, 139, 44,
+        177, 36, 175, 85, 127, 123, 121, 5, 53, 252, 47, 72, 99, 133, 46, 48, 76, 67, 166, 136, 216, 171, 49, 120, 150,
+        197, 94, 234, 36, 251, 59, 102, 43, 196, 54, 55, 138, 254, 248, 226, 27, 75, 64, 65, 70, 179, 143, 249, 27, 85,
+        91, 169, 46, 237, 98, 213, 205, 27,
+    ];
+
+    assert!(matches!(
+        dbg!(Message::unpack_verified(bytes)),
```
Comment (thibault-martinez) [inline bee-message/tests/message.rs]: Was it temporary ? It's going to mess up the output a bit

Comment (pvdrz) [reply bee-message/tests/message.rs]: oh yes it was temporary. I'll remove it

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**